src/main.js

```diff
@@ -340,8 +341,15 @@ const startCreating = async () => {
   let failedCount = 0;
   let abstractedIndexes = [];
   for (
-    let i = network == NETWORK.sol ? 0 : 1;
-    i <= layerConfigurations[layerConfigurations.length - 1].growEditionSizeTo;
+    let i =
+      network == NETWORK.sol
+        ? startEditionFrom > 1
+          ? startEditionFrom
+          : 0
+        : startEditionFrom;
```

Comment:
You got the ternary wrong, 0 should be first.

Also, it would probably be better to have `startEditionFrom` be defined per `layerConfigurations` rather than on it's 
own.

If you do that, instead of checking wether the value is greater than 1, you could use `.startEditionFrom || 1` instead.

You should end up with something like this:
```js
      network == NETWORK.sol
        ? 0
        : (layerConfigurations[layerConfigurations.length - 1].startEditionFrom || 1);
```

I've added this in my PR as well but I doubt it'll get merged.

Reply:
Did this work?

If so, can you please advise where that piece of code belongs?

Reply:
NVM I see where, lines 345-349

Please let me know if this works.

I would really appreciate it.

Reply:
> You got the ternary wrong, 0 should be first.
> 
> Also, it would probably be better to have `startEditionFrom` be defined per `layerConfigurations` rather than on it's own.
> 
> If you do that, instead of checking wether the value is greater than 1, you could use `.startEditionFrom || 1` instead.
> 
> You should end up with something like this:
> 
> ```js
>       network == NETWORK.sol
>         ? 0
>         : (layerConfigurations[layerConfigurations.length - 1].startEditionFrom || 1);
> ```
> 
> I've added this in my PR as well but I doubt it'll get merged.

The reason for the current ternary is to support Solana network too.

`      network == NETWORK.sol
        ? 0
        : (layerConfigurations[layerConfigurations.length - 1].startEditionFrom || 1);`

This will not allow startEditionFrom for solana network.

Reply:
> 

This is working fine for me. I had to create a collection of 3333 NFTs but the collection consisted of 1111 sub-collections. So first I created the collection from 1-1111 then 1112-2222 and finally 2223-3333.

Reply:
> NVM I see where, lines 345-349
> 
> Please let me know if this works.
> 
> I would really appreciate it.

This is working fine for me. I had to create a collection of 3,333 NFTs but it consisted of 1,111 sub-collection. So first I created the sub-collection 1-1111 then 1112-2222 and finally 2223-3333.

Reply:
@jaikpillai
> This will not allow startEditionFrom for solana network.

That's because I don't think solana supports starting from anything but 0. At least I think that, the ternary was there before so I assume it has a reason for it.

If there was no reason for it we could just get rid of the ternary.

Also, in my PR, I load the DNA from the `build/json/_metadata.json` file if it exists. This means there won't be any duplicates in the final collection.

Reply:
> That's because I don't think solana supports starting from anything but 0. At least I think that, the ternary was there before so I assume it has a reason for it.

Yes, that's right token indexing in Solana starts from 0 but the purpose behind `startEditionFrom` for me was to create a collection with various sub-collections. So I wanted a solution where I can generate a set of NFTs from 1-1111 then for the next sub-collection start from 1112-2222 and so on. In my case, the old DNA didn't matter as the layers themselves were different for each sub-collection.

In my PR
If `startEditionFrom=1` and `network == sol` then the collection will start from 0 only (this is to avoid any confusion if the user doesn't want to start from a custom number). But if startEditionFrom is greater than 1, the value of startEditionFrom will be considered.

Reply:
@jaikpillai fair enough, thank you for the explaination.

Reply:
@bolshoytoster your welcome buddy.

Reply:
@bolshoytoster no problem buddy.

Reply:
Hi, plzz help me with this problem I'm getting this error
D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\node_modules\canvas\lib\image.js:91
  SetSource.call(img, src);
            ^

Error: Unsupported image type
    at setSource (D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\node_modules\canvas\lib\image.js:91:13)
    at Image.set (D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\node_modules\canvas\lib\image.js:62:9)
    at D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\node_modules\canvas\index.js:34:15
    at new Promise (<anonymous>)
    at loadImage (D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\node_modules\canvas\index.js:23:10)
    at D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\src\main.js:185:27
    at new Promise (<anonymous>)
    at loadLayerImg (D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\src\main.js:184:12)
    at D:\NFT PROGRAMS\hashlips_art_engine-main\hashlips_art_engine-main\src\main.js:368:31
    at Array.forEach (<anonymous>)
This error appears when it want like at first it appeared at 140 images, then i retry, then it appeared at 250 images, and now 350 
i don't know what to do so I thought i can somehow start the collection from the number it stops, first I decided to copy the created images to another folder and start again and when it will stop again, I will copy the new images and paste them with the previous images but that will work they will name the same and they both will be starting from 1 so they will replace, sorry but please help me, i landed on this page figuring out that how can i start the collection from the stopped number so it will start from the stopped image like if it makes sense, I've paste this code
      network == NETWORK.sol
        ? 0
        : (layerConfigurations[layerConfigurations.length - 1].startEditionFrom || 1);
but that didn't work i've got a syntax error please help me, just send me a screenshot of where you placed it and how

Reply:
@Huzaifa-Zahid your issue has nothing to do with this PR, refer to [these issues](https://github.com/HashLips/hashlips_art_engine/search?q=Error%3A+Unsupported+image+type&type=issues).

Reply:
@bolshoytoster Hi, How can I get the solution for that error any help will be appreciated

Reply:
@Huzaifa-Zahid I don't know, look at the issues I linked.

Reply:
Expanded on @bolshoytoster response a bit:

Line 343:
`  let goUntil = network == NETWORK.sol ? 0 : (layerConfigurations[layerConfigurations.length - 1].startEditionFrom || 1)
  for (
    // let i = network == NETWORK.sol ? 0 : 1;
    let i = goUntil;
    i <= layerConfigurations[layerConfigurations.length - 1].growEditionSizeTo;
    i++
  ) {
    abstractedIndexes.push(i);
  }`

Line 362:
`    while (
      editionCount <= (layerConfigurations[layerConfigIndex].growEditionSizeTo - goUntil + 1)
    ) {`